source/features/show-names.tsx

```diff
@@ -38,7 +38,23 @@ function appendName(element: HTMLAnchorElement, fullName: string): void {
 }
 
 async function updateLinks(found: HTMLAnchorElement[]): Promise<void> {
-	const users = Map.groupBy(found, element => element.textContent.trim());
+	const users = new Map<string, HTMLAnchorElement[]>();
+
+	for (const element of found) {
+		const username = element.textContent.trim();
+
+		// #8661
+		if (username !== element.href.split('/').pop()) {
+			continue;
+		}
+
+		if (users.has(username)) {
+			users.get(username)!.push(element);
+		} else {
+			users.set(username, [element]);
+		}
+	}
+
```

Comment:
```suggestion
	const users = Map.groupBy(
	    found.filter(element => element.textContent.trim() === element.href.split('/').pop()),
	    element => element.textContent.trim()
	);
```
how about this approach?